### PR TITLE
MET-4463 update oai library

### PR DESCRIPTION
**Pull request by jeortizquan**

### CODE REVIEW

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiHarvesterImpl.java [hunk 1]
```diff
@@ -81,10 +81,10 @@ private static ListIdentifiersParameters prepareListIdentifiersParameters(OaiHar
     ListIdentifiersParameters parameters = ListIdentifiersParameters.request()
             .withMetadataPrefix(harvest.getMetadataPrefix());
     if (harvest.getFrom() != null) {
-      parameters.withFrom(Date.from(harvest.getFrom()));
+      parameters.withFrom(Date.from(harvest.getFrom()).toInstant());
```
Comment (stzanakis) [inline metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiHarvesterImpl.java:84]: harvest.getFrom() returns instant already, so the Date.from.toinstant is not required I believe.
Same below.

File: metis-repository/metis-repository-rest/src/main/java/eu/europeana/metis/repository/rest/controller/OaiPmhController.java [hunk 2]
```diff
@@ -37,107 +31,121 @@
...
-        .withHeader(createHeader(oaiRecord)).withMetadata(new Metadata(oaiRecord.getEdmRecord()));
-    return new GetRecord(resultRecord);
-  }
 
-  private static Header createHeader(Record oaiRecord) {
-    final Header result = new Header().withDatestamp(Date.from(oaiRecord.getDateStamp()))
-                                      .withSetSpec(oaiRecord.getDatasetId()).withIdentifier(oaiRecord.getRecordId());
-    if (oaiRecord.isDeleted()) {
-      result.withStatus(Status.DELETED);
+    private class XmlOAIWriterContext implements WriterContext {
```
Comment (stzanakis) [inline metis-repository/metis-repository-rest/src/main/java/eu/europeana/metis/repository/rest/controller/OaiPmhController.java:145]: static perhaps.

File: metis-repository/metis-repository-rest/src/main/java/eu/europeana/metis/repository/rest/controller/OaiPmhController.java [hunk 3]
```diff
@@ -37,107 +31,121 @@
...
+        final OAIPMH result = new OAIPMH().withVerb(verbResult).withResponseDate(new Date().toInstant())
+                .withRequest(
+                        new Request(RestEndpoints.REPOSITORY_OAI_ENDPOINT)
+                                .withVerb(verbResult.getType())
+                );
+        try {
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            final WriterContext context = new XmlOAIWriterContext();
+            XmlWriter writer = new XmlWriter(outputStream, context);
+            try {
```
Comment (stzanakis) [inline metis-repository/metis-repository-rest/src/main/java/eu/europeana/metis/repository/rest/controller/OaiPmhController.java:106]: try-with-resources can be applied.

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiRecordHeader.java [hunk 4]
```diff
@@ -37,7 +37,7 @@ public OaiRecordHeader(String oaiIdentifier, boolean isDeleted, Instant datestam
    */
   static OaiRecordHeader convert(Header header) {
     return new OaiRecordHeader(header.getIdentifier(), header.isDeleted(),
-            Optional.ofNullable(header.getDatestamp()).map(Date::toInstant).orElse(null));
+            Optional.ofNullable(header.getDatestamp()).orElse(null));
```
Comment (stzanakis) [inline metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiRecordHeader.java:40]: This chain is unnecessary? Just use header.getDatestamp()

File: pom.xml [hunk 5]
```diff
@@ -141,7 +141,7 @@
 
     <!-- These two versions are interdependent. -->
     <version.directory.maven.plugin>1.0</version.directory.maven.plugin>
-    <version.dspace.oai>4.2.0</version.dspace.oai>
+    <version.dspace.oai>5.2.0</version.dspace.oai>
```
Comment (stzanakis) [inline pom.xml:144]: Name of the parameter change to accommodate the new library?

**Review by stzanakis — CHANGES_REQUESTED** [review]

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiRecordHeader.java (hunk 4 above)
Comment (jeortizquan) [reply metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiRecordHeader.java:40]: oops. true. done.

File: metis-repository/metis-repository-rest/src/main/java/eu/europeana/metis/repository/rest/controller/OaiPmhController.java (hunk 3 above)
Comment (jeortizquan) [reply metis-repository/metis-repository-rest/src/main/java/eu/europeana/metis/repository/rest/controller/OaiPmhController.java:106]: done.

File: pom.xml (hunk 5 above)
Comment (jeortizquan) [reply pom.xml:144]: done. renamed to `version.gdcc.xoai`

File: metis-repository/metis-repository-rest/src/main/java/eu/europeana/metis/repository/rest/controller/OaiPmhController.java (hunk 2 above)
Comment (jeortizquan) [reply metis-repository/metis-repository-rest/src/main/java/eu/europeana/metis/repository/rest/controller/OaiPmhController.java:145]: done.

File: metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiHarvesterImpl.java (hunk 1 above)
Comment (jeortizquan) [reply metis-harvesting/src/main/java/eu/europeana/metis/harvesting/oaipmh/OaiHarvesterImpl.java:84]: oops. I see. thanks. done.

**Pull request merged**